src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +389,16 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> span)
+        {
+            OutStream.Write(span);
+        }
+
+        public virtual void Write(ReadOnlySpan<char> span)
+        {
+            OutStream.Write(span.NonPortableCast<char, byte>());
```

Comment:
This needs to a) take the same approach I outlined in my previous comment, but also b) use encodings the same way the char[] overload does.